### PR TITLE
feat: the opensea token is-tweet bid, the token should be brought into the opensea token.

### LINKED ISSUE #5383: [Crash] Error: [BigNumber Error] Argument not an integer: NaN
I was try to swap from clicking "get more DAI" button, then Mask reports an error.

> [BigNumber Error] Argument not an integer: NaN

Error stack:

<pre>Error: [BigNumber Error] Argument not an integer: NaN
    at intCheck (chrome-extension://ogllhbgmamcggllocglncccbknonfcna/js/npm.bignumber.js.js:8474:13)
    at BigNumber.P.shiftedBy (chrome-extension://ogllhbgmamcggllocglncccbknonfcna/js/npm.bignumber.js.js:7999:7)
    at rightShift (chrome-extension://ogllhbgmamcggllocglncccbknonfcna/js/229.js:778:73)
    at chrome-extension://ogllhbgmamcggllocglncccbknonfcna/js/chunk.315.js:2722:54
    at Object.qi [as useMemo] (chrome-extension://ogllhbgmamcggllocglncccbknonfcna/js/npm.react-dom.js:183:119)
    at exports.useMemo (chrome-extension://ogllhbgmamcggllocglncccbknonfcna/js/2000.js:4950:297)
    at MakeOfferDialog (chrome-extension://ogllhbgmamcggllocglncccbknonfcna/js/chunk.315.js:2719:48)
    at Mh (chrome-extension://ogllhbgmamcggllocglncccbknonfcna/js/npm.react-dom.js:172:137)
    at Zi (chrome-extension://ogllhbgmamcggllocglncccbknonfcna/js/npm.react-dom.js:218:256)
    at Gk (chrome-extension://ogllhbgmamcggllocglncccbknonfcna/js/npm.react-dom.js:297:202)</pre>

## Build info
- Version: 2.3.0
- NODE_ENV: production
- target: chromium
- build: beta
- architecture: web
- BUILD_DATE: 2022-01-04T13:18:13.402Z
- VERSION: v1.29.12-1679-g9106240c0(lelenei/mask-518)

## Git (*):

9106240c0 (lelenei/mask-518) on tag "v2.1.0"
git@github.com:DimensionDev/Maskbook.git

### PR DESCRIPTION
**Pull request by lelenei**
## Description

Closes #5383

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
  - [x] I have removed all in development `console.log`s
  - [x] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

**Comment by UncleBill** [comment]
Make the title and the commit message more detail. You can add `fix mask-518` as a description message of the commit.


```sh
git commit -m "fix: what problem in detail" -m "fix mask-518"
```

**Review by Jack-Works — DISMISSED** [review]

**Comment by orange187** [comment]
Checked

File: packages/mask/src/plugins/Collectible/SNSAdaptor/SelectTokenListPanel.tsx [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+                    placeholder="0.0"
+                    inputProps={{
+                        autoComplete: 'off',
+                        autoCorrect: 'off',
+                        title: 'Token Amount',
+                        inputMode: 'decimal',
+                        min: 0,
+                        minLength: MIN_AMOUNT_LENGTH,
+                        maxLength: MAX_AMOUNT_LENGTH,
+                        pattern: '^[0-9]*[.,]?[0-9]*$',
```
Comment (septs) [inline packages/mask/src/plugins/Collectible/SNSAdaptor/SelectTokenListPanel.tsx]:
```suggestion
                        pattern: /^\d+[.,]?\d+$/.source,
```

File: packages/mask/src/plugins/Collectible/SNSAdaptor/MakeOfferDialog.tsx [hunk 2]
```diff
@@ -139,7 +138,7 @@ export function MakeOfferDialog(props: MakeOfferDialogProps) {
 
     const validationMessage = useMemo(() => {
         setInsufficientBalance(false)
-        const amount_ = rightShift(amount, token.value?.decimals)
+        const amount_ = rightShift(amount ?? '0', Number.isNaN(token.value?.decimals ?? 0) ? 0 : token.value?.decimals)
```
Comment (septs) [inline packages/mask/src/plugins/Collectible/SNSAdaptor/MakeOfferDialog.tsx]:
```suggestion
        const amount_ = rightShift(amount ?? '0', token.value?.decimals || 0)
```

```
> Number.NaN || 0
0
> undefined || 0
0
> null || 0
0
> Number.NaN ?? 0
NaN
```

**Review by Jack-Works — APPROVED** [review]

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**